### PR TITLE
Feat/bc chart

### PR DESCRIPTION
**Pull request by tspoff**
Squash and merge this one due to poor commit names

### CODE REVIEW

File: src/components/BondingCurveChart.tsx [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        const generated = generateChart();
+        data = generated.data;
+        options = generated.options;
+    }
+
+    /*
+        Draw a strait line for the inital goal. The PRICE for this is the slope for total supply 0
+        We may have to hardcode this value...
+
+        It's a slope but I DON"T understand because it's 1 / 10^18!!!! And it seems to be about 1/3 of tokens.
```
Comment (jpkcambridge) [inline src/components/BondingCurveChart.tsx:377]: still relevant?

File: src/components/BondingCurveChart.tsx [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+                    </ChartHeaderFullElement>
+                </ChartBox>
+            </ChartHeaderWrapper>
+            <ChartWrapper>
+                {requiredDataLoaded ? (
+                    <Line
+                        data={data}
+                        options={options}
+                        // width={1000}
+                        // height={250}
```
Comment (jpkcambridge) [inline src/components/BondingCurveChart.tsx:416]: Can we remove these?

File: src/stores/Token.ts [hunk 3]
```diff
@@ -334,6 +334,8 @@ export default class TokenStore {
             responses.forEach((response) => {
                 if (response instanceof TotalSupplyFetch) {
                     const { status, request, payload } = response;
+                    console.log(response);
```
Comment (jpkcambridge) [inline src/stores/Token.ts:337]: log statement

File: src/components/BondingCurveChart.tsx [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+                    [points.curveStart, points.maxSupplyToShow],
+                    chartGray,
+                    'Curve chart funded'
+                )
+            );
+        }
+
+        datasets.push(generateSupplyMarker(points.currentSupply, 'Current Supply'));
+
+        // TODO: Determine how to display predicted future supply / price after current buy
```
Comment (jpkcambridge) [inline src/components/BondingCurveChart.tsx:352]: We probably don't need this as a TODO.  Can add it as a nice to have enhancement under the github issues.

**Comment by jpkcambridge** [comment]
@tspoff Lots of log statements.  Also can squash these commits, probably to one

File: src/components/BondingCurveChart.tsx (hunk 1 above)
Comment (tspoff) [reply src/components/BondingCurveChart.tsx:377]: Yea... I'll stop writing notes in the code

File: src/components/BondingCurveChart.tsx (hunk 2 above)
Comment (tspoff) [reply src/components/BondingCurveChart.tsx:416]: Removed

File: src/stores/Token.ts (hunk 3 above)
Comment (tspoff) [reply src/stores/Token.ts]: Removed

File: src/components/BondingCurveChart.tsx (hunk 4 above)
Comment (tspoff) [reply src/components/BondingCurveChart.tsx]: I'm going to just include the feature for now - It'll help us think about the best way to display it. Created an issued as well.

**Comment by AugustoL** [comment]
Can we also update the main stats above the chart in this PR?

**Pull request merged**